src/site/content/en/blog/optimize-cls/index.md

```diff
@@ -16,32 +16,113 @@ tags:
+{% endAside %}
+
+### Identifying load CLS issues
+
+When the CrUX and Lighthouse CLS scores of PageSpeed Insights are broadly in line, this indicates there is a load CLS issue that was detected by Lighthouse. In this case Lighthouse will help with two audits to provide more information on images causing CLS due to missing width and height, and also list all the elements that shifted for the page load along with their CLS contribution.
+
+{% Img src="image/W3z1f5ZkBJSgL1V1IfloTIctbIF3/2C3v6dGwPx2yFyYpVdue.png", alt="Lighthouse Screenshot showing the CLS audits providing more information to help you identify and address CLS issues", width="800", height="544" %}
+
+{% Aside important %}
+  Lighthouse will identify the elements that were shifted, but often these are the ones _impacted_ rather than the elements _causing_ the CLS. For example, if a new element is inserted into the DOM the elements that are beneath it will show in this audit, but the root cause is the addition of the new element above. Hopefully, the shifted element will be sufficient to help you identify and resolve the root cause.
+{% endAside %}
+
+### Identifying non-load CLS issues
+
+When the CrUX and Lighthouse CLS scores of PageSpeed Insights are not in line, then this likely indicates post-load CLS. To replicate that, you can browse your web page while [recording CLS using a JavaScript snippet](/cls/#measure-cls-in-javascript) pasted into the console.
+
+Alternatively, the [Web Vitals Chrome extension](https://chrome.google.com/webstore/detail/web-vitals/ahfhijdlegdabablpippeagghigmibma) can be used to monitor CLS as you interact with a page, either in a heads up display, or in the console.
+
+The [Performance panel](https://developer.chrome.com/docs/devtools/evaluate-performance/) in DevTools highlights layout shifts in the **Experience** section. The **Summary** view for a `Layout Shift` record includes the cumulative layout shift score as well as a rectangle overlay showing the affected regions.
+
+<figure>
+  {% Img src="image/admin/ApDKifKCRNGWI2SXSR1g.jpg", alt="Layout Shift records being displayed in the Chrome DevTools performance panel when expanding the Experience section", width="800", height="438" %}
+  <figcaption>After recording a new trace in the Performance panel, the <b>Experience</b> section of the results is populated with a red-tinted bar displaying a <code>Layout Shift</code> record. Clicking the record allows you to drill down into impacted elements (e.g. note the moved from/to entries).</figcaption>
+</figure>
+
+The newer [Performance Insights panel](https://developer.chrome.com/docs/devtools/performance-insights/) also includes full support for identifying shifts.
+
+### Measuring CLS elements in the field
```

Comment:
Expanded on the middle section a bit:

>The [attribution functionality of the web-vitals library](https://github.com/GoogleChrome/web-vitals#send-attribution-data) allows this additional information to be collected. Read our [Debug performance in the field](/debug-performance-in-the-field/) post for more information on how to do this. Other RUM providers have also started collecting and presenting this data similarly.

I think the flow of "here's how to identify, here's how to get your users to identify" works better. Also is the pattern we should really be encouraging to CLS issues are caught before making it to the field.

Reply:
The preceding section on post-load CLS lists all the different ways to diagnose post-load issues, but it doesn't make clear which tool(s) to use for the job or when. And given that this section follows it, a reader wouldn't yet know what they're supposed to be doing in these tools to find the post-load issues.

So I still think it'd be helpful to rearrange the sections, so that we set the stage with collecting CLS attribution info in the field, then talk about using that info to reproduce layout shifts experienced by real users. That way it's a bit more methodical than interacting with the page to see what happens.

Reply:
I disagree. I think users are unlikely to measure in the field before they know they have a problem. And once they have a problem they will be unlikely to wait while sufficient field data is collected to wait to fix it.

To me, adding field data is you moving to a more advanced level of managing your CLS. But most users are likely to start without it. So the progression from trying to identify it first yourself, seeing how painful that is, and then measuring in field as a solution to that seems a natural order.

Those users who already are collecting the field attribution data, have likely already been through this, or know to jump straight to this data as they know they are collecting it.

Reply:
> The preceding section on post-load CLS lists all the different ways to diagnose post-load issues, but it doesn't make clear which tool(s) to use for the job or when. And given that this section follows it, a reader wouldn't yet know what they're supposed to be doing in these tools to find the post-load issues.

I've change this a bit more to remove the Performance Panel tracing section (I've moved it to the previous Load CLS issue section as believe it's more useful there) and also added a section on common post-load CLS causes.

I still believe the RUM section follows more naturally in the next section, as a solution to the "guesswork" nature of this. Especially as, like I said above, anyone that's already collecting this in RUM likely knows to go there (at least until this attribution data becomes a lot more common in RUM solutions).